src/app/components/index/index.component.ts

```diff
@@ -145,11 +150,52 @@ export class IndexComponent implements OnInit {
   }
 
   handleCompetitionDeleteClick(competitionId: number) {
-    this.competitionService.deleteCompetition(competitionId).subscribe(() => {
-      this.competitionService.getCompetitions().subscribe((competitions) => {
-        this.competitions = competitions;
-        this.selectCompetitions();
-      });
+    const dialogRef = this.dialog.open(ConfirmDialogComponent, {
+      data: {
+        title: 'Delete competition',
+        content: 'Are you sure you want to delete this competition?',
+      },
+    });
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result) {
+        this.competitionService
+          .deleteCompetition(competitionId)
+          .subscribe(() => {
+            this.competitionService
+              .getCompetitions()
+              .subscribe((competitions) => {
+                this.competitions = competitions;
+                this.selectCompetitions();
+              });
+          });
+      }
+    });
+  }
+
+  handleEventViewClick(eventId: number) {
```

Comment:
methot at line 175 and 179 can be turned into 1 method that has 2 params: the route and the evnt id